src/compiler/checker.ts

```diff
@@ -1251,6 +1251,15 @@ namespace ts {
                         patternAmbientModuleAugmentations.set((moduleName as StringLiteral).text, merged);
                     }
                     else {
+                        if (mainModule.exports?.get(InternalSymbolName.ExportStar) && moduleAugmentation.symbol.exports && moduleAugmentation.symbol.exports.size) {
+                            // We may need to merge the module augmentation's exports into the target symbols of the resolved exports
+                            const resolvedExports = getResolvedMembersOrExportsOfSymbol(mainModule, MembersOrExportsResolutionKind.resolvedExports);
+                            for (const [key, value] of arrayFrom(moduleAugmentation.symbol.exports.entries())) {
+                                if (resolvedExports.has(key) && !mainModule.exports.has(key)) {
+                                    mergeSymbol(resolvedExports.get(key)!, value);
+                                }
+                            }
+                        }
                         mergeSymbol(mainModule, moduleAugmentation.symbol);

```

Comment:
remind me: what does it mean to still `mergeSymbol(mainModule, moduleAugmentation.symbol)` after manually merging the resolved export star? Are they mutually exclusive, so both need to be done?

Reply:
the manual merging is merging anything that exists in the export star and the augmentation into the export star target, the full ns merge here ensures that any augmentations that introduce new exports are then picked up and added.